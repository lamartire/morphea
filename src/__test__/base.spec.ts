import { css, createStylesFromString } from '../index'

it('createStylesFromString â€“ creates styles object from string', () => {
  expect(createStylesFromString('display:block;width:100px;')).toMatchSnapshot()
})

describe('css', () => {
  it('returns empty object', () => {
    expect(css``).toMatchSnapshot()
  })

  it('creates styles object', () => {
    expect(css`
      display: block;
      width: 100px;
      color: #fff;
      background: rgba(255, 255, 255, 0);
    `).toMatchSnapshot()
  })

  it('creates styles with variables', () => {
    expect(css`
      display: ${'block'};
      width: ${'100px'};
      color: ${'#fff'};
      background: ${'rgba(255,255,255,0)'};
    `).toMatchSnapshot()
  })

  it('merge other css in template literals', () => {
    expect(css`
      display: block;
      width: ${100}px;
      height: 100px;
      ${css`
        color: #f00;
      `};
    `).toMatchSnapshot()
  })
})
