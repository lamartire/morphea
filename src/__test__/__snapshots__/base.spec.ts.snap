// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createStylesFromString â€“ creates styles object from string 1`] = `
Object {
  "display": "block",
  "width": "100px",
}
`;

exports[`css creates styles object 1`] = `
Object {
  "background": "rgba(255,255,255,0)",
  "color": "#fff",
  "display": "block",
  "width": "100px",
}
`;

exports[`css creates styles with variables 1`] = `
Object {
  "background": "rgba(255,255,255,0)",
  "color": "#fff",
  "display": "block",
  "width": "100px",
}
`;

exports[`css merge other css in template literals 1`] = `
Object {
  "display": "block",
  "height": "100px",
  "width": "100px",
}
`;

exports[`css returns empty object 1`] = `Object {}`;
